server/src/main/java/org/elasticsearch/action/fieldcaps/FieldCapabilitiesResponse.java

```diff
@@ -190,4 +230,19 @@ public int hashCode() {
     public String toString() {
         return Strings.toString(this);
     }
+
+    @Override
+    public RestStatus status() {
+        RestStatus status = RestStatus.OK;
+        if (indices.length == 0 && failures.size() > 0) {
```

Comment:
Do we need this logic now that we throw an exception if all indices fail ? We could even assert in the ctr that failures  must be empty if indices is empty.